File: services/api/internal/grpcserver/health.go
package grpcserver

import (
	"context"

	pb "github.com/irfndi/match-bot/packages/contracts/gen/go/proto/meetsmatch/v1"
)

type HealthService struct {
	pb.UnimplementedHealthServiceServer
}

func (s *HealthService) Check(ctx context.Context, req *pb.HealthCheckRequest) (*pb.HealthCheckResponse, error) {
	return &pb.HealthCheckResponse{Status: "ok"}, nil
}

File: packages/cf-shared/src/contracts/health.ts
import {
  Array,
  Boolean,
  Literal,
  Number,
  String,
  Struct,
  optional,
} from "effect/Schema";

export const HealthCheckRequest = Struct({});
export type HealthCheckRequest = typeof HealthCheckRequest.Type;

export const HealthCheckResponse = Struct({
  status: String,
});
export type HealthCheckResponse = typeof HealthCheckResponse.Type;

export interface HealthService {
  readonly check: (req: HealthCheckRequest) => Promise<HealthCheckResponse>;
}
